palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/service/PalisadeService.java

```diff
@@ -15,43 +15,76 @@
 
     /**
-     * This method is used by the client code to register that they want to read a resource or data set. This method
-     * will check that the user can have access to the resource and pass back details of all the resources linked to the
-     * initial request (if they asked for a data set) and how to connect to the relevant data service to get that data.
-     * Then behind the scenes this will have triggered more services to be called to collate the data required by the
-     * getDataRequestConfig() method so the data service can apply the relevant policies.
+     * Create token string.
      *
-     * @param request a {@link RegisterDataRequest} that contains the details the client needs to provide the palisade
-     *                service for it to register the data request
-     * @return details of all the resources linked to the initial request (if they asked for a data set) and how to
-     * connect to the relevant data service to get that data.
+     * @param palisadeRequest the original request
+     * @return the string
      */
-    DataRequestResponse registerDataRequest(RegisterDataRequest request);
+    public abstract String createToken(PalisadeRequest palisadeRequest);
 
     /**
-     * This method is used by the data service's to request the trusted details that it requires to apply the necessary
-     * data access controls.
+     * This method will forward the data to a Kafka data stream where it can be retrieve by the user-service.
+     * The incoming request is in the form of a {@link PalisadeRequest} which contains all of the information provided by the
+     * client for this registering this data request. Service will include a unique token to identify this data request and
+     * the data relevant to the request.
      *
-     * @param request This is the {@link GetDataRequestConfig} that the client passed to the data service.
-     * @return a {@link DataRequestConfig} containing the information that the data service requires to apply the
-     * necessary filtering/transformations to the data.
+     * @param request information about the data, user requesting the data and the context of the request.
+     * @return the token for later accessing the results of the request at the data-service.
      */
-    DataRequestConfig getDataRequestConfig(GetDataRequestConfig request);
+    public CompletableFuture<String> registerDataRequest(final PalisadeRequest request) {
+        // needs to send the information to the stream
+        // what is needed is to include the token and the Palisade request as part of the source
+        String token = this.createToken(request);
+        TokenRequestPair incomingMessage = new TokenRequestPair(token, request);
+
+        return futureSink.thenApply((Sink<TokenRequestPair, ?> sink) -> {
+            Source.single(incomingMessage).runWith(sink, materializer);
+            LOGGER.debug("registerDataRequest returning with token {}, and request {}", token, request);
```

Comment:
```suggestion
            LOGGER.debug("registerDataRequest returning with token {} for request {}", token, request);
```
I think this makes more sense as the request is not returned by the method.